asset/partition_by_fields/processor.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const {
+    BatchProcessor
+} = require('@terascope/job-components');
+const path = require('path');
+
+class PartitionByFields extends BatchProcessor {
+    addPath(record, opConfig) {
+        const partitions = [];
+        opConfig.fields.forEach((field) => {
+            partitions.push(`${field}=${record[field]}`.replace(/\//gi, '_'));
```

Comment:
We should probably replace `=` to avoid breaking the partitioning
```suggestion
            partitions.push(`${field}=${record[field]}`.replace(/[=\/]/g, '_'));
```

Reply:
Good catch. I needed to add this to `record[field]` in order to prevent it from wiping out the intended `=`, but I fixed that in the hash partitioner and added tests for fields with `=` in both